vnext/Scripts/copyRNLibraries.js

```diff
@@ -4,121 +4,115 @@
 // @ts-check
 const path = require('path');
 const fs = require('fs');
+const glob = require('glob');
 const rimraf = require('rimraf');
 
-function copyFileSync(source, target) {
-  var targetFile = target;
+function retryOnError(errorCode, func) {
+  let shouldLoop = true;
 
-  //if target is a directory a new file with the same name will be created
-  if (fs.existsSync(target)) {
-    if (fs.lstatSync(target).isDirectory()) {
-      targetFile = path.join(target, path.basename(source));
+  while (shouldLoop) {
+    try {
+      func();
+      shouldLoop = false;
+    } catch (ex) {
+      if (ex.code !== errorCode) {
+        throw ex;
+      }
     }
   }
-
-  fs.writeFileSync(targetFile, fs.readFileSync(source));
 }
 
-function copyJSFolderRecursiveSync(source, target) {
-  var files = [];
-
-  //check if folder needs to be created or integrated
-  var targetFolder = path.join(target, path.basename(source));
-  if (!fs.existsSync(targetFolder)) {
-    fs.mkdirSync(targetFolder);
-  }
-
-  //copy
-  if (fs.lstatSync(source).isDirectory()) {
-    files = fs.readdirSync(source);
-    files.forEach(function(file) {
-      var curSource = path.join(source, file);
-      if (fs.lstatSync(curSource).isDirectory()) {
-        copyJSFolderRecursiveSync(curSource, targetFolder);
-      } else {
-        if (
-          curSource.endsWith('.js') ||
-          curSource.endsWith('.png') ||
-          curSource.endsWith('.gif')
-        ) {
-          copyFileSync(curSource, targetFolder);
+function copyDirectories(srcPath, targetPath, dirSpecs) {
+  dirSpecs.forEach(dirSpec => {
+    if (dirSpec.mergeFiles !== true) {
+      // rimraf issue 72: We will see ENOTEMPTY if Windows is still holding a
+      // lock to a file in our directory. E.g. if we get unlucky with timing
+      // and antivirus decides to run. Keep on spinning until we're able to
+      // delete the directory.
+      retryOnError('ENOTEMPTY', () => {
```

Comment:
From what I can tell we managed to avoid this issue previously by pure luck. My guess is either timing or some internal logic used by the synchronous node APIs to check folder existence caused us to not see these errors. I was seeing them on some CI runs before adding this, but never locally. Blocking a thread by spinning without telling the user isn't really ideal, but my hope is that it shouldn't have a real impact.

Reply:
Maybe we should have it print out that its retrying, and maybe fail if it retries 10 times or something?

Reply:
I can add a time based abort, since I'm guessing iterations happen quickly. I was originally wanting to do some sort of backoff but discovered node doesn't have a built in way to sleep, making that hard to do synchronously without adding dependencies.

Reply:
Dont do anything that complicates things too much.  -- certainly not worth adding any dependencies for.